### PR TITLE
Lock Scrim After 3 Rejections

### PR DESCRIPTION
**Pull request by bwpikaard**

### CODE REVIEW

File: microservices/matchmaking-service/src/scrim/scrim.event-subscriber.ts [hunk 1]
```diff
@@ -100,6 +106,11 @@ export class ScrimEventSubscriber {
             this.logger.warn(`Scrim not found for submission ${payload.submissionId}`);
             return;
         }
-        await this.scrimService.resetScrim(scrim.id);
+        await this.scrimService.resetScrim(scrim.id).catch(() => null);
```
Comment (bwpikaard) [inline microservices/matchmaking-service/src/scrim/scrim.event-subscriber.ts:109]: Caused instances of "scrim was already in x state" without catching. @0c370t look into?

Comment (0c370t) [reply microservices/matchmaking-service/src/scrim/scrim.event-subscriber.ts:109]: Not sure where that specific error would come from; nothing in this function does that, except for the check that the scrim is in ratifying; which _should_ be the case at this point. Not a huge fan of just swallowing the error for the sake of swallowing it.

File: microservices/submission-service/src/replay-submission/replay-submission-crud/replay-submission-crud.controller.ts [hunk 2]
```diff
@@ -15,4 +15,11 @@ export class ReplaySubmissionCrudController {
         await this.crudService.removeSubmission(data.submissionId);
         return {removed: true};
     }
+
+    @MessagePattern(SubmissionEndpoint.GetSubmissionRejections)
+    async getSubmissionRejections(@Payload() payload: unknown): Promise<IGetSubmissionRejections_Response> {
```
Comment (0c370t) [inline microservices/submission-service/src/replay-submission/replay-submission-crud/replay-submission-crud.controller.ts:20]: I think there is a way to do this, without needing to create an interface; but I could be wrong. Lets look into it on Sunday.

**Review by 0c370t — CHANGES_REQUESTED** [review]

File: microservices/matchmaking-service/src/scrim/scrim.event-subscriber.ts (hunk 1 above)
Comment (bwpikaard) [reply microservices/matchmaking-service/src/scrim/scrim.event-subscriber.ts]: That seem better?

File: microservices/submission-service/src/replay-submission/replay-submission-crud/replay-submission-crud.controller.ts (hunk 2 above)
Comment (0c370t) [reply microservices/submission-service/src/replay-submission/replay-submission-crud/replay-submission-crud.controller.ts:20]: `SubmissionOutput<SubmissionEndpoint.GetSubmissionRejections>`

**Pull request merged**